### ISSUE TITLE
Problem with Isbg

### ISSUE THREAD

**Issue by Jaska80** [isbg/isbg#126]
Hi everyone.

My wife gets tons on spam in her email and i have used isbg on my rpi3 for couple of years to clean them out

However now i get this kind of error?
it checks most off emails just like it should but eventually i get this error.

Traceback (most recent call last):
  File "/usr/local/bin/isbg", line 11, in <module>
    load_entry_point('isbg==2.1.5', 'console_scripts', 'isbg')()
  File "/usr/local/lib/python3.7/site-packages/isbg-2.1.5-py3.7.egg/isbg/__main__.py", line 262, in main
    return sbg.do_isbg()  # return the exit code.
  File "/usr/local/lib/python3.7/site-packages/isbg-2.1.5-py3.7.egg/isbg/isbg.py", line 600, in do_isbg
    proc = self.do_spamassassin()
  File "/usr/local/lib/python3.7/site-packages/isbg-2.1.5-py3.7.egg/isbg/isbg.py", line 521, in do_spamassassin
    proc = sa.process_inbox(origpastuids)
  File "/usr/local/lib/python3.7/site-packages/isbg-2.1.5-py3.7.egg/isbg/spamproc.py", line 439, in process_inbox
    logger=self.logger)
  File "/usr/local/lib/python3.7/site-packages/isbg-2.1.5-py3.7.egg/isbg/imaputils.py", line 143, in get_message
    mail = new_message(body)
  File "/usr/local/lib/python3.7/site-packages/isbg-2.1.5-py3.7.egg/isbg/imaputils.py", line 96, in new_message
    if mail.as_bytes() in [b'', b'\n']:
  File "/usr/local/lib/python3.7/email/message.py", line 178, in as_bytes
    g.flatten(self, unixfrom=unixfrom)
  File "/usr/local/lib/python3.7/email/generator.py", line 116, in flatten
    self._write(msg)
  File "/usr/local/lib/python3.7/email/generator.py", line 181, in _write
    self._dispatch(msg)
  File "/usr/local/lib/python3.7/email/generator.py", line 214, in _dispatch
    meth(msg)
  File "/usr/local/lib/python3.7/email/generator.py", line 264, in _handle_multipart
    self.write(subparts)
  File "/usr/local/lib/python3.7/email/generator.py", line 406, in write
    self._fp.write(s.encode('ascii', 'surrogateescape'))
UnicodeEncodeError: 'ascii' codec can't encode characters in position 3639-3641: ordinal not in range(128)

Im definitely no expert with linux so please help me out. 
Am i doing something wrong?

**Comment by baldurmen** [comment]
Could you try to tell me from what email provider (Gmail, Yahoo, etc.) the email ISBG crashed on is?

You can see that when running ISBG with the `--verbose-mails` option.

I suspect it's some badly formatted email causing the problem, as I've never seen that before.

**Comment by Jaska80** [comment]
Hi,

Email provider is Elisa, It's a local finnish provider.

Took a little time to find out what message it was that crashed isbg and now we have a winner "Amazing russian for you" :)

Received: via tmail-2007f.2015-sau for s******.300; Mon, 2 Sep 2019 12:24:10 +0300 (EEST)
Received: from fe23.mail.saunalahti.fi (fe23.mail.saunalahti.fi [62.142.5.28])
    by be517.mail.saunalahti.fi (Postfix) with ESMTP id 300C960568
    for <s******@be517.mail.saunalahti.fi>; Mon, 2 Sep 2019 12:24:10 +0300 (EEST)
X-Client-Addr: 172.106.173.5
Received: from marriage0exist.com (unknown [172.106.173.5])
    by fe23.mail.saunalahti.fi (Postfix) with ESMTP id 07A4120014
    for <h*****a.s*****o@netikka.fi>; Mon, 2 Sep 2019 12:24:10 +0300 (EEST)
From: RussiaWomenOnline.com <admin@existing.purpleanthem.online>
List-Unsubscribe: <>
Subject: amazing Russian for you
Date: Mon, 02 Sep 2019 09:16:32 +0000
Reply-To: <8zz@existing.purpleanthem.online>
Content-Type: multipart/mixed;
    boundary="----=_NextPart_6B34F3F8_4354_464B_B214_CE288821B8D1"

**Comment by baldurmen** [comment]
Thanks for the help. I've been looking at the code and I feel this is indeed a bug in the ISBG code.

For some reasons, the message's body is treated as a `bytes` object ([passes this if statement](https://github.com/isbg/isbg/blob/master/isbg/imaputils.py#L93)) but isn't, as `as_bytes()` fails on it since some part of it is ascii.

I'm going to blame multipart/mixed madness for that.

I've pushed a potential fix to [this new branch](https://github.com/isbg/isbg/tree/issue_126). Could you try to test that? I don't know how familiar you are with Linux, but if you don't want to install this manually, I guess just replicating [this small change](https://github.com/isbg/isbg/commit/9dd088ed57d6e07bcabd660b4bd7aeb8171cf084) should be enough.

Would it be possible to attach the full .eml file here with personal infos redacted from the header? I'd like to make some tests on my side and possible use that email in our test suite if I'm right.

**Comment by Jaska80** [comment]
Hi,

I modified that line but didnt work for me.
Or maybe i just dont know how :) 

Here is the email
[amazing Russian for you.zip](https://github.com/isbg/isbg/files/3613451/amazing.Russian.for.you.zip)

Thanks again for your great work with software, Wife get's about 100 spam emails per day so cant live without this.

**Comment by baldurmen** [comment]
I tested ISBG with the email you sent me and I've been unable to reproduce it. I do feel like the patch I made would solve your problem though.

I'll polish up my patch to fix the test suite failure and I'll merge it. Meanwhile, I would recommend you delete the email causing problem manually. That should fix your immediate problem.

Thanks for reporting bugs, it's really important and helps us a lot!!
